src/power_grid_model/core/serialization.py

```diff
@@ -94,10 +100,16 @@ class Serializer(ABC):
     _dataset: CConstDataset
     _serializer: SerializerPtr
 
-    def __new__(cls, data: Dataset, serialization_type: SerializationType, dataset_type: Optional[DatasetType] = None):
+    def __new__(cls, data: Dataset, serialization_type: SerializationType, dataset_type: DatasetType | None = None):
```

Comment:
Yup, no good reason I did it. Its just a personal preference to not have an additional keyword here

Reply:
I would guess that since we had a PR that involved moving forward towards the use of `|`, it would be better to keep it, but if we don't have any concrete guidelines, either should be fine.